plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/SortByInputModelProcessor.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.elk.alg.layered.intermediate;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.elk.alg.layered.graph.LEdge;
+import org.eclipse.elk.alg.layered.graph.LGraph;
+import org.eclipse.elk.alg.layered.graph.LNode;
+import org.eclipse.elk.alg.layered.graph.LPort;
+import org.eclipse.elk.alg.layered.graph.Layer;
+import org.eclipse.elk.alg.layered.options.InternalProperties;
+import org.eclipse.elk.alg.layered.options.LayeredOptions;
+import org.eclipse.elk.core.alg.ILayoutProcessor;
+import org.eclipse.elk.core.options.PortConstraints;
+import org.eclipse.elk.core.util.IElkProgressMonitor;
+
+/**
+ * Sorts all child nodes and their edges of the given graph by their {@code MODEL_ORDER},
+ * which represents the order in input graph.
+ * Outgoing ports are sorted by the order of their edges, incoming ports are sorted by the order of their nodes in the
+ * previous layer.
```

Comment:
Why the different treatment?

Reply:
Because fixing both produces conflicting cases that do not seem to be conflicting:

Given the following graph

```
node n1
edge n1 -> n2
edge n1 -> n3

node n2
edge n2 -> n3
node n3

```
I guess we would assume that this is not conflicting and produces a drawing such as this:

![Screenshot from 2020-06-16 14-40-37](https://user-images.githubusercontent.com/6419799/84775263-477a8100-afdf-11ea-865c-881f7ace4d85.png)

However, if we also fix the order of the incoming ports it is conflicting, since n1 -> n3 has a lower order than n2-> n3.

Incoming ports are not directly stated in the file, only outgoing ports (at least this is how we do it in SCCharts).